api/src/app/modules/project/project.service.ts

```diff
@@ -503,6 +506,28 @@ export class ProjectService extends DataService<Project, Repository<Project>, Pr
         throw new BadRequestException(`Unable to transition FOM ${entity.id} to ${stateTransition}.  
         Proposed submission is required.`);
       }
+
+      // Checks postDate on or before commenting open date and 1 day later than today (before published)
+      // However, it could be empty = then no check.
+      const publicNotices = entity.publicNotices;
+      if (!_.isEmpty(publicNotices)) {
+        const postDate = publicNotices[0].postDate;
+        if (!_.isEmpty(postDate)) {
+					const commentingOpenDate = entity.commentingOpenDate;
+					if (postDate && !DateTimeUtil.isPNPostdateOnOrBeforeCommentingOpenDate(postDate, commentingOpenDate)) {
+						throw new BadRequestException(`Unable to transition FOM ${entity.id} to ${stateTransition}. 
+						Online Public Notice post date ${postDate} should be on or before commenting start date 
+						${commentingOpenDate}.`);
+					}
+					// Must be at least one day after publish is pushed
+					const dayDiff = DateTimeUtil.diffNow(postDate, DateTimeUtil.TIMEZONE_VANCOUVER, 'day');
+					if (dayDiff < 1) {
+							throw new BadRequestException(`Unable to transition FOM ${entity.id} to ${stateTransition}.  
+							Public Notice Post Date: must be at least one day after publish is pushed.`);
```

Comment:
I think the terminology of these error messages needs to match what Julius requested for the terminology on the screen "Notice Publish Date" or something like that?